src/Components/Selection.re

```diff
@@ -0,0 +1,53 @@
+[@deriving show({with_path: false})]
+type t = {
+  anchor: int,
+  focus: int,
+};
+
+let initial: t = {anchor: 0, focus: 0};
+
+let anchor = (selection: t): int => {
+  selection.anchor;
+};
+
+let focus = (selection: t): int => {
+  selection.focus;
+};
+
+let withinLength = (text: string, position: int): int => {
+  let withinStart = max(position, 0);
+  let withinBoth = min(withinStart, String.length(text));
+
+  withinBoth;
+};
```

Comment:
We have a utility function for this already:

https://github.com/onivim/oni2/blob/b7238298362d4ecce8402739fc2e377e043edf11/src/Core/Utility/IntEx.re#L1-L11